src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -446,14 +447,15 @@ static int open_device(const char* path, bool try_load_moudle = false)
 {
   int nbd = open(path, O_RDWR);
   if (nbd < 0 && try_load_moudle && access("/sys/module/nbd", F_OK) != 0) {
+    ostringstream param;
     int r;
     if (nbds_max) {
-      ostringstream param;
       param << "nbds_max=" << nbds_max;
-      r = module_load("nbd", param.str().c_str());
-    } else {
-      r = module_load("nbd", NULL);
     }
+    if (max_part) {
```

Comment:
This looks could be more useful if a user could specify max_part via command line, like now it is possible  to specify nbd_max via '--nbds_max' (actually, not sure how useful it is, but at least we would be consistent with nbds_max). Also, it might make sense to change the condition to `if (max_part > 0)` to avoid negative value input.

@liupan1111 are you interested in improving this?